lib/database_connection.rb

```diff
@@ -0,0 +1,43 @@
+# file: lib/database_connection.rb
+
+require 'pg'
+
+# This class is a thin "wrapper" around the
+# PG library. We'll use it in our project to interact
+# with the database using SQL.
+
+class DatabaseConnection
+  # This method connects to PostgreSQL using the 
+  # PG gem. We connect to 127.0.0.1, and select
+  # the database name given in argument.
+  def self.connect
+
+    if ENV['DATABASE_URL'] != nil
+      @connection = PG.connect(ENV['DATABASE_URL'])
+      return
+    end
+
+    if ENV['ENV'] == 'test'
```

Comment:
This feels surplus to requirements given that both conditional paths have the same outcome (though I understand why you've kept this in so as not to screw with the ENV variable set-up